app/controllers/KeepAliveController.scala

```diff
@@ -27,15 +27,15 @@ import scala.concurrent.{ExecutionContext, Future}
 class KeepAliveController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   authorise: AuthoriseAction,
-  getData: DataRetrievalAction,
+  getData: SessionRequestAction,
   sessionRepository: SessionRepository
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController {
 
   def keepAlive: Action[AnyContent] = (authorise andThen getData).async { implicit request =>
-    request.userAnswers
-      .map { answers =>
-        sessionRepository.keepAlive(answers.id).map(_ => Ok)
+    request.userAnswers.traderGoodsProfile.ukimsNumber
```

Comment:
why are we passing the ukims number here?

Reply:
Oh I see, I think previously userAnswers was optional in the request whereas it's not optional only need to call the keep alive function here and it seems that the way it's handling errors is by ignoring them..returning an Ok.

It does raise the question though, shouldn't we be using the session service  here, but that would require more work we we'd need to handle errors appropriately (at the moment exceptions are not being caught)

Reply:
This is from Ewans branch I think :)

Reply:
my bad if theres an issue here, I had a lot to juggle